File: Sprint7/A/C++/main.cpp
// https://contest.yandex.ru/contest/25596/problems/A/
#include <iostream>
#include <vector>

int main() {
    int n;

    std::cin >> n;
    std::vector<int> prices(n);

    for (int i = 0; i < n; ++i) {
        std::cin >> prices[i];
    }

    if (n <= 1) {
        std::cout << 0 << std::endl;
        return 0;
    }

    int max_profit = 0;

    for (int i = 1; i < n; i++) {
        if (prices[i] > prices[i - 1])
            max_profit += prices[i] - prices[i - 1];
    }

    std::cout << max_profit << std::endl;

    return 0;
}

File: Sprint7/A/Rust/src/main.rs
// https://contest.yandex.ru/contest/25596/problems/A/
fn main() {
    let mut lines = std::io::stdin().lines();
    let n = lines.next().unwrap().unwrap().parse::<usize>().unwrap();

    let prices = lines
        .next()
        .unwrap()
        .unwrap()
        .split_whitespace()
        .map(|x| x.parse::<usize>().unwrap())
        .collect::<Vec<usize>>();

    if n <= 1 {
        println!("0");
        return;
    }

    let mut max_profit = 0;

    for i in 1..n {
        if prices[i] > prices[i - 1] {
            max_profit += prices[i] - prices[i - 1];
        }
    }

    println!("{}", max_profit);
}
